com.unity.visualeffectgraph/Editor/Controls/VFXMinMaxSliderField.cs

```diff
@@ -0,0 +1,199 @@
+            }
+
+            void OnMouseDown(MouseDownEvent e)
+            {
+                target.RegisterCallback<MouseMoveEvent>(OnMouseMove, TrickleDown.TrickleDown);
+            }
+
+            void OnMouseMove(MouseMoveEvent e)
+            {
+                if (!m_InDrag)
+                {
+                    m_InDrag = true;
+                    m_Slider.ValueDragStarted();
+                    target.UnregisterCallback<MouseMoveEvent>(OnMouseMove, TrickleDown.TrickleDown); //we care only about the first drag event
+                }
+            }
+
+            void OnMouseUp(MouseUpEvent e)
+            {
+                if (m_InDrag)
+                    m_Slider.ValueDragFinished();
+                else
+                    target.UnregisterCallback<MouseMoveEvent>(OnMouseMove, TrickleDown.TrickleDown);
+                m_InDrag = false;
+            }
+        }
+
+        protected void ValueDragFinished()
+        {
+            if (onValueDragFinished != null)
+                onValueDragFinished();
+        }
+
+        protected void ValueDragStarted()
+        {
+            if (onValueDragStarted != null)
+                onValueDragStarted();
+        }
+
+        public Action onValueDragFinished;
+        public Action onValueDragStarted;
+
+        public VFXMinMaxSliderField()
+        {
+            AddToClassList("sliderMinMaxField");
+            m_Slider = new MinMaxSlider(1, 10, 0, 100);
+
+            m_Slider.RegisterValueChangedCallback(evt => ValueChanged(evt.newValue));
+            m_IndeterminateLabel = new Label()
+            {
+                name = "indeterminate",
+                text = VFXControlConstants.indeterminateText
+            };
+            m_IndeterminateLabel.SetEnabled(false);
+            m_Slider.RegisterValueChangedCallback(ValueChanged);
+
+            Add(m_Slider);
+            RegisterCallBack();
+        }
+
+        protected void RegisterCallBack()
+        {
+            m_Slider.Children().First().AddManipulator(new StartFinishSliderManipulator());
+        }
+
+        void OnFocusLost(BlurEvent e)
+        {
+            //forward the focus lost event
+            using (BlurEvent newE = BlurEvent.GetPooled(this, e.relatedTarget, e.direction, panel.focusController))
+            {
+                SendEvent(newE);
+            }
+
+            e.StopPropagation();
+        }
+
+        public Vector2 m_Value;
+
+        public Vector2 value
+        {
+            get => m_Value;
+            set => SetValueAndNotify(value);
+        }
+
+        private Vector2 m_Range;
+
+        public Vector2 range
+        {
+            get
+            {
+                return m_Range;
+            }
+            set
+            {
+                m_Range = value;
+                m_IgnoreNotification = true;
+                if (m_Slider.lowLimit != m_Range.x || m_Slider.highLimit != m_Range.y)
```

Comment:
If you check that values are different before assign them I guess that's to avoid useless process?
If so, I think there could be useless updates because you are comparing float value.

Reply:
It's possible, but the values don't come from a computation, so I don't expect issues coming from rounding.